ethers-providers/src/pubsub.rs

```diff
@@ -78,20 +78,17 @@ where
     P: PubsubClient,
     R: DeserializeOwned,
 {
-    type Item = R;
+    type Item = Result<R, serde_json::Error>;
```

Comment:
this will break a lot

instead we should rather create a new `TrySubscriptionStream` type that returns results.

this would require some refactoring but so we can reuse lots of code but is the cleaner solution imo